ports/stm32/Makefile

```diff
@@ -347,6 +347,10 @@ SRC_USBDEV = $(addprefix $(USBDEV_DIR)/,\
 	class/src/usbd_msc_data.c \
 	)
 
+ifeq ($(MICROPY_HW_CLK_SRC_HSI),1)
+CFLAGS_MOD += -DMICROPY_HW_CLK_SRC_HSI=1
+endif
```

Comment:
It doesn't look like `MICROPY_HW_CLK_SRC_HSI` needs to be something you set via the Makefile (eg it doesn't include/exclude certain .c files in the build).  For options that don't need to be via the Makefile it's best to keep them just internal to C.  Basically these lines are not needed.